JsonParser.cpp

```diff
@@ -0,0 +1,65 @@
+#include "JsonParser.h"
+#include "Exceptions.h"
+
+#include <sstream>
+#include <fstream>
+
+
+std::map<std::string, std::string> JsonParser::parseJson(std::istream & instream)
+{
+	std::map<std::string, std::string> content;
+	std::string line;
+
+	while (std::getline(instream, line))
+	{
+		size_t position;
+		size_t colonPos;
+		position = line.find("\"");
+		if (position == std::string::npos)	continue;
+		colonPos = line.find(":");
+		if (colonPos == std::string::npos) throw InputFormatException(":");
+
+		//getting the key
+		std::string key = line.substr(position + 1);
+		position = key.find("\"");
+		if(position > key.find(":")) throw InputFormatException("\"");
+		if (position == std::string::npos)	 throw InputFormatException("\"");
+		key = key.substr(0, position);
+
+		//getting the value
+		std::string value = line.substr(colonPos + 1);
+		value = value.substr(value.find_first_not_of(" \t\v\""));
+		position = value.find_last_not_of(" \t\v,\"");
+		value = value.substr(0, position + 1);
+
+		content.insert(std::pair<std::string, std::string>(key, value));
+
+	}
+	
+	return content;
+}
+
+std::map<std::string, std::string> JsonParser::parseJson(std::string & fname)
```

Comment:
Miért kevered a fájlnévvel történő munkát a string tartalmának feldolgozásával? Legyen erre két külön fv.!

Reply:
@wajzy 
Azért keverem, mert mindegyiket stringként kapom, így két fv-nek is ugyan az lenne a szignatúrája.
Itt az lenne a megoldás, hogy miután eldöntöttem, hogy nem fájlnév a kapott `fname `string, meghívok egy harmadik függvényt, ami majd elvégzi ami egyelőre itt van?
Vagy már eleve külön függvényt kellene meghívjak rá (amit az azonos szignatúrák miatt nem tudom hogyan csináljak)?

Reply:
Miért kellene, hogy a szignatúra azonos legyen? Nevezd át az egyik függvényt! Pontosan tudni fogod, mikor melyiket kell meghívni, hiszen tudod, milyen bemenetet adsz majd neki.